x-pack/plugins/maps/public/classes/joins/inner_join.ts

```diff
@@ -70,12 +88,15 @@ export class InnerJoin {
           featurePropertyKey.length >= stylePropertyPrefix.length &&
           featurePropertyKey.substring(0, stylePropertyPrefix.length) === stylePropertyPrefix
         ) {
-          delete feature.properties[featurePropertyKey];
+          // For some reason, ts compilation complains about this deletion
+          if (feature.properties) {
+            delete feature.properties[featurePropertyKey];
```

Comment:
Need to use `not null` operator here. Typescript does not recognize your check on line 75 because its not in scope of the anonymous function containing this use of feature.properties.